conda_forge_tick/update_upstream_versions.py

```diff
@@ -187,38 +188,45 @@ def get_version(self, url):
         return url
 
 
-def get_latest_version(meta_yaml, sources):
-    for source in sources:
-        url = source.get_url(meta_yaml)
-        if url is None:
-            continue
-        ver = source.get_version(url)
-        if ver:
-            return ver
-        else:
-            meta_yaml["bad"] = "Upstream: Could not find version on {}".format(
-                source.name
-            )
-    if not meta_yaml.get("bad"):
-        meta_yaml["bad"] = "Upstream: unknown source"
-    return False
+def get_latest_version(name, meta_yaml, sources):
+    try:
+        for source in sources:
+            url = source.get_url(meta_yaml)
+            if url is None:
+                continue
+            ver = source.get_version(url)
+            if ver:
+                meta_yaml["new_version"] = ver
+                return name, meta_yaml
+            else:
+                meta_yaml["bad"] = "Upstream: Could not find version on {}".format(
+                    source.name
+                )
+        if not meta_yaml.get("bad"):
+            meta_yaml["bad"] = "Upstream: unknown source"
+        meta_yaml["new_version"] = False
+        return name, meta_yaml
+    except Exception as e:
+        logger.warn("Error getting uptream version of {}: {}".format(name, e))
+        meta_yaml["bad"] = "Upstream: Error getting upstream version"
+        meta_yaml["new_version"] = False
+        return name, meta_yaml
 
 
 def update_upstream_versions(gx, sources=(PyPI(), CRAN(), RawURL(), Github())):
-    for node, attrs in gx.node.items():
-        if attrs.get("bad") or attrs.get("archived"):
-            attrs["new_version"] = False
-            continue
-        try:
-            attrs["new_version"] = get_latest_version(attrs, sources)
-        except Exception as e:
-            logger.warn("Error getting uptream version of {}: {}".format(node, e))
-            attrs["bad"] = "Upstream: Error getting upstream version"
-            attrs["new_version"] = False
-        else:
-            logger.info(
-                "{} - {} - {}".format(node, attrs["version"], attrs["new_version"])
-            )
+    futures = []
+    with ThreadPoolExecutor(max_workers=20) as pool:
+        for node, attrs in gx.node.items():
+            if attrs.get("bad") or attrs.get("archived"):
+                attrs["new_version"] = False
+                continue
+            futures.append(pool.submit(get_latest_version(attrs, sources)))
+        for f in as_completed(futures):
+            node, attrs = f.result()
```

Comment:
I think you can put the try here instead of inside the functions.

Reply:
Will I have access to the name for the logger?